cmd/snap-update-ns/utils.go

```diff
@@ -89,31 +89,32 @@ func (sec *Secure) CheckTrespassing(fd int, segments []string, segNum int) error
 // The file descriptor is opened using the O_PATH, O_NOFOLLOW,
 // and O_CLOEXEC flags.
 func (sec *Secure) OpenPath(path string) (int, error) {
-	if !filepath.IsAbs(path) {
-		return -1, fmt.Errorf("path %v is not absolute", path)
-	}
-	segments, err := splitIntoSegments(path)
+	iter, err := NewPathIterator(path)
 	if err != nil {
-		return -1, err
+		// TODO: remove this and adjust tests
+		return -1, fmt.Errorf("cannot split unclean path %q", path)
+	}
+	iter.Next()
+	if iter.CurrentName() != "/" {
```

Comment:
You could do the check before calling iter.Next():
```
if filepath.IsAbs(iter.Path()) {
    return -1, fmt.Errorf("path %v is not absolute", path)
}
```

Reply:
I agree with @bboozzoo, but I think it should be:
```
if !filepath.IsAbs(iter.Path()) {
    return -1, fmt.Errorf("path %v is not absolute", iter.Path())
}
```

Reply:
Right, `if !filepath.IsAbs().. {` silly typo :)

Reply:
I'm checking this now